### PR TITLE
Add MediaJobOutputProgressEventData in Media Services EventGrid swagger

### PR DESCRIPTION
**Pull request by jinshang-msft**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by anuchandy** [comment]
@kalyanaj could you take a look given this is an integration between event grid and media?

File: specification/eventgrid/data-plane/Microsoft.Media/stable/2018-01-01/MediaServices.json [hunk 1]
```diff
@@ -366,6 +366,29 @@
       "type": "object",
       "description": "The event data for a Job output asset."
     },
+    "MediaJobOutputProgressEventData": {
+      "x-ms-discriminator-value": "#Microsoft.Media.JobOutputProgress",
```
Comment (kalyanaj) [inline specification/eventgrid/data-plane/Microsoft.Media/stable/2018-01-01/MediaServices.json]: Didn't quite understand why the discriminator is needed for this. Can you please clarify?

Comment (Chuansssss) [reply specification/eventgrid/data-plane/Microsoft.Media/stable/2018-01-01/MediaServices.json]: The value of x-ms-discriminator-value is how the client and server differentiate polymorphic types, which, in our case, the type of the event. You could find similar attribute in other event data like MediaJobScheduledEventData and MediaJobOutputFinishedEventData.

Comment (kalyanaj) [reply specification/eventgrid/data-plane/Microsoft.Media/stable/2018-01-01/MediaServices.json]: Got it, thanks.

**Review by kalyanaj — APPROVED** [review]
LGTM. Once this get merged, will you also be able to help update event types list/event mapping/tests for Media services event types in the C# SDK to include this new event type?  The relevant links are:

https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/EventGrid/DataPlane/Microsoft.Azure.EventGrid/Customization/EventTypes.cs, 
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/EventGrid/DataPlane/Microsoft.Azure.EventGrid/Customization/SystemEventTypeMappings.cs, 
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/EventGrid/DataPlane/Microsoft.Azure.EventGrid.Tests/Tests/ConsumeEventTests.cs

**Comment by Chuansssss** [comment]
Updated event types in azure-sdk-for-net in this [commit](https://github.com/Azure/azure-sdk-for-net/compare/psSdkJson6...Chuansssss:ams_update_mediajoboutputprogress)

**Review by anuchandy — APPROVED** [review]

**Comment by anuchandy** [comment]
thanks @kalyanaj, @Chuansssss.

**Pull request merged**